src/test/unit/java/godaddy/registry/jtoolkit2/se/idn/ietf/IetfIdnCommandExtensionTest.java

```diff
@@ -0,0 +1,170 @@
+package godaddy.registry.jtoolkit2.se.idn.ietf;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import godaddy.registry.jtoolkit2.Timer;
+import godaddy.registry.jtoolkit2.se.CLTRID;
+import godaddy.registry.jtoolkit2.se.Command;
+import godaddy.registry.jtoolkit2.se.DomainCheckCommand;
+import godaddy.registry.jtoolkit2.se.DomainCreateCommand;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.xml.sax.SAXException;
+
+public class IetfIdnCommandExtensionTest {
+
+    public static final String DOMAIN_NAME = "jtkutest.com.au";
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Before
+    public void setUp() throws Exception {
+        Timer.setTime("20070101.010101");
+        CLTRID.setClID("JTKUTEST");
+    }
+
+    @Test
+    public void testDomainCreateIdnCommandWithTableAndUname() {
+        Command cmd = new DomainCreateCommand(DOMAIN_NAME, "jtkUT3st");
+        IetfIdnCommandExtension idnExt = new IetfIdnCommandExtension("es", "uname");
+        cmd.appendExtension(idnExt);
+
+        String expectedExtension = new StringBuilder("<data xmlns=\"urn:ietf:params:xml:ns:idn-1.0\">")
+                .append("<table>es</table>")
+                .append("<uname>uname</uname>")
+                .append("</data>")
+                .toString();
+        try {
+            String xml = cmd.toXML();
+            assertEquals(expectedDomainCreateXmlWithExtension(expectedExtension), xml);
+        } catch (SAXException saxe) {
+            fail(saxe.getMessage());
+        }
+    }
+
+    @Test
+    public void testDomainCreateIdnCommandWithTableButWithoutUname() {
+        Command cmd = new DomainCreateCommand(DOMAIN_NAME, "jtkUT3st");
+        IetfIdnCommandExtension idnExt = new IetfIdnCommandExtension("es", null);
+        cmd.appendExtension(idnExt);
+        String expectedExtension = new StringBuilder("<data xmlns=\"urn:ietf:params:xml:ns:idn-1.0\">")
+                .append("<table>es</table>")
+                .append("</data>")
+                .toString();
+        try {
+            String xml = cmd.toXML();
+            assertEquals(expectedDomainCreateXmlWithExtension(expectedExtension), xml);
+        } catch (SAXException saxe) {
+            fail(saxe.getMessage());
+        }
+    }
+
+    @Test
+    public void failCreateCommandWhenTableNotProvided() throws SAXException {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Table must not be null or empty.");
+        Command cmd = new DomainCreateCommand(DOMAIN_NAME, "jtkUT3st");
+        IetfIdnCommandExtension idnExt = new IetfIdnCommandExtension(null, null);
+        cmd.appendExtension(idnExt);
+        cmd.toXML();
+    }
```

Comment:
This test case has unnecessary setup.

As we know, the exception was thrown from the constructor of the renamed `IetfIdnCommandExtension`. There is no need to create command objects (for either CREATE or CHECK command) because it would never run to these lines:

```java
        cmd.appendExtension(idnExt);
        cmd.toXML();
```

Below code illustrate what I mean - please note, I have not run/test it locally.

```suggestion
    @Test
    public void failIetfIdnExtensionWhenTableNotProvided() throws SAXException {
        thrown.expect(IllegalArgumentException.class);
        thrown.expectMessage("Table must not be null or empty.");
        IetfIdnCommandExtension idnExt = new IetfIdnCommandExtension(null, null);
    }
```

Understood this test case was not authored by you, but there is a similar one introduced for domain check. We just need to keep one test case, as it's not dependent on `Command` at all.

Reply:
Done